src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -72,88 +86,290 @@ def close(self):
+            #self.checkStatus(request)
 
     def stop(self):
-        "Stop RequestManager"
+        "Stop MSManager"
+        # stop MSTransferorMonit thread
+        self.ms_monit.stop()
+        # stop MSTransferor thread
         self.thr.stop() # stop checkStatus thread
         status = self.thr.running()
         return status
 
-    def checkPhedex(self, request):
+    def transferRequest(self, request, rdict):
+        "Send request to Phedex and return status of request subscription"
+	datasets = rdict.get('datasets', [])
+	sites = rdict.get('sites', [])
+        if datasets and sites:
+	    subscription = PhEDExSubscription(datasets, sites, self.group)
+	    self.logger.debug("### add subscription %s for request %s" % (subscription, request))
+            # TODO: implement how to get transfer id
+            tid = hashlib.md5(str(subscription)).hexdigest()
+	    # TODO: when ready enable submit subscription step
+	    # self.phedex.subscribe(subscription)
+            return tid
+
+    def getTransferIDsDoc(self):
+        """
+        Get transfer ids document from backend. The document has the following form:
+        https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+	{"wf_A": {"timestamp": 0000
+		  "primary": {"dset_1": ["list of transfer ids"]},
+		  "secondary": {"PU_dset_1": ["list of transfer ids"]},
+	 "wf_B": {"timestamp": 0000
+		  "primary": {"dset_1": ["list of transfer ids"],
+			      "parent_dset_1": ["list of transfer ids"]},
+		  "secondary": {"PU_dset_1": ["list of transfer ids"],
+				"PU_dset_2": ["list of transfer ids"]},
+         ...
+	}
+        """
+        doc = {}
+        return doc
+
+    def updateTransferIDs(self, trecords):
+        "Update transfer ids in backend"
+        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
+        doc = self.getTransferIDsDoc()
+        pass
+
+    def getTransferIDs(self):
+        "Get transfer ids from backend"
+        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
+        # meanwhile return transfer ids from internal store 
+        tids = [r['tid'] for r in self.store.records()]
+        return tids
+
+    def getRequest(self, tid):
+        "Return request name for given transfer id"
+        records = self.store.records()
+        for rec in records:
+            if rec['tid'] == tid:
+                return rec['request']
+
+    def getTransferStatuses(self, tids):
+        "get transfer ids from backend"
+        # transfer docs on backend has the following form
+        # https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+        statuses = []
+        records = self.store.records()
+        store_tids = [r.get('tid', 0) for r in records]
+        for tid in tids:
+            if tid in store_tids:
+                for rec in records:
+                    if rec.get('tid', 0) == tid:
+                       statuses.append(rec['status'])
+                       break
+            else:
```

Comment:
For the record, this else should never get executed. The MS is supposed to persist all transfer ids, so if we cannot find it, we have a bug.

Reply:
once again, it is placeholder code to make whole workflow working. Once proper APIs will be in place it will be re-written. Moreover, most likely I'll remove internal store as well.